src/ray/common/client_connection.cc

```diff
@@ -602,4 +607,32 @@ std::string ServerConnection::DebugString() const {
   return result.str();
 }
 
+std::vector<bool> CheckForClientDisconnects(
+    const std::vector<std::shared_ptr<ClientConnection>> &conns) {
+  std::vector<bool> result(conns.size(), false);
+#if defined(_WIN32)
+  return result;
+#else
+  // Poll for POLLHUP on all of the FDs in a single syscall.
+  std::vector<pollfd> poll_fds(conns.size());
+  for (size_t i = 0; i < conns.size(); ++i) {
+    poll_fds[i] = {conns[i]->GetNativeHandle(), POLLHUP, 0};
```

Comment:
```
POLLHUP
              Hang up (only returned in revents; ignored in events).
              Note that when reading from a channel such as a pipe or a
              stream socket, this event merely indicates that the peer
              closed its end of the channel.  Subsequent reads from the
              channel will return 0 (end of file) only after all
              outstanding data in the channel has been consumed.
```

Seems POLLHUP is always checked and setting it in events will simply be ignored.

Reply:
oh nice catch

Reply:
updated with a comment